alicloud/table_alicloud_ecs_eni.go

```diff
@@ -0,0 +1,290 @@
+				Hydrate:     getEcsEniAka,
+				Transform:   transform.FromValue(),
+			},
+			{
+				Name:        "title",
+				Description: ColumnDescriptionTitle,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("NetworkInterfaceName"),
+			},
+
+			// alibaba standard columns
+			{
+				Name:        "region",
+				Description: ColumnDescriptionRegion,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("ZoneId").Transform(zoneToRegion),
+			},
+			{
+				Name:        "account_id",
+				Description: "The alicloud Account ID in which the resource is located.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getCommonColumns,
+				Transform:   transform.FromField("AccountID"),
+			},
+		},
+	}
+}
+
+//// LIST FUNCTION
+
+func listEcsEni(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	region := plugin.GetMatrixItem(ctx)[matrixKeyRegion].(string)
+	// Create service connection
+	client, err := ECSService(ctx, d, region)
+	if err != nil {
+		plugin.Logger(ctx).Error("alicloud_ecs_eni.listEcsEni", "connection_error", err)
+		return nil, err
+	}
+	request := ecs.CreateDescribeNetworkInterfacesRequest()
+	request.Scheme = "https"
+	request.PageSize = requests.NewInteger(50)
+	request.PageNumber = requests.NewInteger(1)
+
+	count := 0
+	for {
+		response, err := client.DescribeNetworkInterfaces(request)
+		if err != nil {
+			plugin.Logger(ctx).Error("alicloud_ecs_eni.listEcsEni", "query_error", err, "request", request)
+			return nil, err
+		}
+		for _, eni := range response.NetworkInterfaceSets.NetworkInterfaceSet {
+			plugin.Logger(ctx).Warn("listEcsEni", "item", eni)
+			d.StreamListItem(ctx, eni)
+			count++
+		}
+		if count >= response.TotalCount {
+			break
+		}
+		request.PageNumber = requests.NewInteger(response.PageNumber + 1)
+	}
+	return nil, nil
+}
+
+//// GET FUNCTION
+
+func getEcsEni(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	region := plugin.GetMatrixItem(ctx)[matrixKeyRegion].(string)
+	plugin.Logger(ctx).Trace("getEcsEni")
+
+	// Create service connection
+	client, err := ECSService(ctx, d, region)
+	if err != nil {
+		plugin.Logger(ctx).Error("alicloud_ecs_eni.getEcsEni", "connection_error", err)
+		return nil, err
+	}
+
+	var id string
+	if h.Item != nil {
+		NetworkInterfaceSet := h.Item.(ecs.NetworkInterfaceSet)
+		id = NetworkInterfaceSet.NetworkInterfaceId
+	} else {
+		id = d.KeyColumnQuals["network_interface_id"].GetStringValue()
+	}
+
+	request := ecs.CreateDescribeNetworkInterfaceAttributeRequest()
+	request.Scheme = "https"
+	request.NetworkInterfaceId = id
+
+	response, err := client.DescribeNetworkInterfaceAttribute(request)
+	if serverErr, ok := err.(*errors.ServerError); ok {
+		plugin.Logger(ctx).Error("alicloud_ecs_eni.getEcsEni", "query_error", serverErr, "request", request)
+		return nil, serverErr
+	}
+
+	return response, nil
+}
```

Comment:
Use `DescribeNetworkInterfaces` API for GET, and filter result using `NetworkInterfaceId`.
Add a separate hydrate function for `DescribeNetworkInterfaceAttribute` API.